tasks/sep/metal-001-MCP-L1/SignFromJson.s.sol

```diff
@@ -0,0 +1,448 @@
+    uint256 constant l2OutputOracleSubmissionInterval = 180; // cast call 0x75a6B961c8da942Ee03CA641B09C322549f6FA98 "SUBMISSION_INTERVAL()(uint256)"
+    uint256 constant finalizationPeriodSeconds = 604800; // cast call 0x75a6B961c8da942Ee03CA641B09C322549f6FA98 "FINALIZATION_PERIOD_SECONDS()(uint256)
+    uint256 constant l2OutputOracleStartingTimestamp = 1708129620; // cast call 0x75a6B961c8da942Ee03CA641B09C322549f6FA98 "startingTimestamp()(uint)"
+    uint256 constant l2OutputOracleStartingBlockNumber = 0; // TODO is this ok? cast call 0x75a6B961c8da942Ee03CA641B09C322549f6FA98 "startingBlockNumber()(uint)"
+    uint256 constant l2GenesisBlockGasLimit = 30e6;
+    uint256 constant requiredProtocolVersion = 0x0000000000000000000000000000000000000006000000000000000000000003; // cast call 0x79ADD5713B383DAa0a138d3C4780C7A1804a8090 "required()(bytes32)"
+    uint256 constant recommendedProtocolVersion = 0x0000000000000000000000000000000000000006000000000000000000000003; // cast call 0x79ADD5713B383DAa0a138d3C4780C7A1804a8090 "recommended()(bytes32)"
+    uint256 constant xdmSenderSlotNumber = 204; // Verify against https://github.com/ethereum-optimism/optimism/blob/e2307008d8bc3f125f97814243cc72e8b47c117e/packages/contracts-bedrock/snapshots/storageLayout/L1CrossDomainMessenger.json#L93-L99
+
+    Types.ContractSet proxies;
+
+    /// @notice Sets up the contract
+    function setUp() public {
+        proxies = _getContractSet();
+    }
+
+    function checkSemvers() internal view {
+        // These are the expected semvers based on the `op-contracts/v1.3.0` release.
+        // https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2fv1.3.0
+        require(ISemver(proxies.L1CrossDomainMessenger).version().eq("2.3.0"), "semver-100");
+        require(ISemver(proxies.L1StandardBridge).version().eq("2.1.0"), "semver-200");
+        require(ISemver(proxies.L2OutputOracle).version().eq("1.8.0"), "semver-300");
+        require(ISemver(proxies.OptimismMintableERC20Factory).version().eq("1.9.0"), "semver-400");
+        require(ISemver(proxies.OptimismPortal).version().eq("2.5.0"), "semver-500");
+        require(ISemver(proxies.SystemConfig).version().eq("1.12.0"), "semver-600");
+        require(ISemver(proxies.L1ERC721Bridge).version().eq("2.1.0"), "semver-700");
+        require(ISemver(proxies.ProtocolVersions).version().eq("1.0.0"), "semver-800");
+        require(ISemver(proxies.SuperchainConfig).version().eq("1.1.0"), "semver-900");
+    }
+
+    /// @notice Asserts that the SystemConfig is setup correctly
+    function checkSystemConfig() internal view {
+        console.log("Running assertions on the SystemConfig");
+
+        require(proxies.SystemConfig.code.length != 0, "200");
+        require(EIP1967Helper.getImplementation(proxies.SystemConfig).code.length != 0, "201");
+
+        SystemConfig systemConfig = SystemConfig(proxies.SystemConfig);
```

Comment:
What do we need to modify to the chain assertions library that we use in the monorepo to be able to reuse them here?

Reply:
It would help reduce overhead as we can maintain a single source of truth for assertions about the contracts

Reply:
This is a good question and something @maurelian asked also. I'd like to hold off and see the upcoming deploy-config proposal from @protolambda and see before spending too much time thinking about this, since any current solution will likely need to change

Reply:
Ok that sounds good to me